src/main/groovy/com/ofg/twitter/place/extractor/PlacesExtractor.groovy

```diff
@@ -5,28 +5,36 @@ import com.ofg.twitter.place.model.Tweet
 import groovy.transform.PackageScope
 import groovy.util.logging.Slf4j
 import groovyx.gpars.GParsPool
+import org.springframework.cloud.sleuth.Span
+import org.springframework.cloud.sleuth.Trace
+import org.springframework.cloud.sleuth.TraceContextHolder
+import org.springframework.cloud.sleuth.TraceScope
 
 import java.util.concurrent.ConcurrentHashMap
 
-import static com.ofg.infrastructure.correlationid.CorrelationIdUpdater.wrapClosureWithId
-
 @PackageScope
 @Slf4j
 class PlacesExtractor {
 
     private final List<PlaceExtractor> placeExtractors
     private final Meter analyzedTweetsMeter
+    private final Trace trace
 
-    PlacesExtractor(List<PlaceExtractor> placeExtractors, Meter analyzedTweetsMeter) {
+    PlacesExtractor(List<PlaceExtractor> placeExtractors, Meter analyzedTweetsMeter, Trace trace) {
         this.placeExtractors = placeExtractors
         this.analyzedTweetsMeter = analyzedTweetsMeter
+        this.trace = trace
     }
 
     Map<String, Optional<Place>> extractPlacesFrom(List<Tweet> tweets) {
         Map<String, Optional<Place>> foundPlaces = new ConcurrentHashMap<>()
+        Span span = TraceContextHolder.currentSpan
         GParsPool.withPool {
-            tweets.eachParallel wrapClosureWithId { tweet ->
+            tweets.eachParallel { tweet ->
+                TraceScope traceScope = trace.startSpan(Thread.currentThread().name, span)
                 foundPlaces << appendExtractedTweet(tweet)
+                traceScope.close()
+                TraceContextHolder.removeCurrentSpan()
```

Comment:
We could add a similar static method to achieve a similar result. The problem is that you need to pass also the span. We're not actively using GPars in our projects so I'd ignore that for the time being (Remember that in micro-infra we want to one day have everything written in Java so it will not understand what a Closure is).